### ISSUE TITLE
Allow option to pin dependencies

### ISSUE THREAD

**Issue by doug-wade** [cedx/david.gulp#9]
Pinned dependencies can be very useful for certain use cases; for instance, we'd like to minimize continuous integration failures due to issues like https://github.com/eslint/eslint/issues/5680.  Using david is even more important for projects that use pinned dependencies, since its very tedious to run `npm outdated` and copy-paste all the suggested versions.  It'd be great to have an option, pin, that changes the behavior of update to update dependencies to the latest pinned version, rather than a caret range.

Examples:

```
return gulp.src('package.json')
  .pipe(david({ update: true, pin: true }))
  .pipe(gulp.dest('.'));
```

outputs 

package.json
```
dependencies: {
    "chalk": "2.2.0"
}
```

```
return gulp.src('package.json')
  .pipe(david({ update: true }))
  .pipe(gulp.dest('.'));
```

outputs 

package.json
```
dependencies: {
    "chalk": "^2.2.0"
}
```

```
return gulp.src('package.json')
  .pipe(david({ update: true, pin: true }))
  .pipe(gulp.dest('.'));
```

outputs 

stdout.out
```
chalk 2.2.0
```

**Comment by cedx** [comment]
I think this is a good idea. I will try to implement this feature in the next days.

**Comment by cedx** [comment]
I will implement this feature as an option specifying the operator to use (or not use) in version comparators. Something like `updateOperator: String = '^'`.

This let you use the operator of your choice, not only the tilde and caret ones.
What do you think of this proposal ? Do you think that `updateOperator` is enough meaningful?

**Comment by doug-wade** [comment]
I like the idea of allowing you to set any of the node semver ranges -- it seems like a much more flexible implementation of my original feature request.  I think that you'd want to default to `^`, but that has the strange property that to pin dependencies you'd have config like

**Comment by cedx** [comment]
The equality operator (=) can be used to pin dependencies:
https://docs.npmjs.com/misc/semver#ranges

> `=` Equal. If no operator is specified, then equality is assumed, so this operator is optional, but MAY be included.

I will finally merge `update` and `updateOperator` options. The behavior of the `update` option will be:
- `false`: will not update.
- `true`: will update using the caret operator (`^`)
- String : will update using the specified operator. To pin the dependencies, provide the `=` string.

**Comment by doug-wade** [comment]
Awesome!

**Comment by cedx** [comment]
Implemented in version 0.4.2 (to be released soon !).
